src/main/scala/beam/router/skim/core/AbstractSkimmer.scala

```diff
@@ -157,12 +160,7 @@ abstract class AbstractSkimmer(beamConfig: BeamConfig, ioController: OutputDirec
       case e: AbstractSkimmerEvent if e.getEventType == eventType =>
         currentSkimInternal.compute(
           e.getKey,
-          (_, v) => {
-            val value =
```

Comment:
You can just return from the `if`/`else`. Or else I'd suggest a better name than `value`. As it stands I do not see much benefit to the `val`, as it would usually be for readability here.

Reply:
I actually removed that `val value`. Option(null) creates None which allowed me to remove the `if/else`.